From 2a3fe9a75951cb085b81569939f6af3ce2eb2b02 Mon Sep 17 00:00:00 2001
From: Kay Sievers <kay@vrfy.org>
Date: Fri, 20 Jul 2012 16:01:13 +0200
Subject: [PATCH 1/2] udev: path_id - export path for platform block devices

Enrico Scholz <enrico.scholz@sigma-chemnitz.de>
>
> E.g. I have a platform with two sdhci controllers with different purposes.
> First slot is an external slot while second one is internal with a
> non-removable card.
>
> When there is a card in the external slot at boot, the non-removable card is
> named 'mmcblk1'; without the external card it is 'mmcblk0'.  Vice versa for the
> external card.

https://bugs.freedesktop.org/show_bug.cgi?id=52309
---
 src/udev/udev-builtin-path_id.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/udev/udev-builtin-path_id.c b/src/udev/udev-builtin-path_id.c
index 223d9b1..8eac991 100644
--- a/src/udev/udev-builtin-path_id.c
+++ b/src/udev/udev-builtin-path_id.c
@@ -458,6 +458,7 @@ static int builtin_path_id(struct udev_device *dev, int argc, char *argv[], bool
                 } else if (strcmp(subsys, "platform") == 0) {
                         path_prepend(&path, "platform-%s", udev_device_get_sysname(parent));
                         parent = skip_subsystem(parent, "platform");
+                        some_transport = true;
                 } else if (strcmp(subsys, "acpi") == 0) {
                         path_prepend(&path, "acpi-%s", udev_device_get_sysname(parent));
                         parent = skip_subsystem(parent, "acpi");
-- 
1.7.11.2

